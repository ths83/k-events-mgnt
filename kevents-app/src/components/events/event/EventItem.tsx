import List from "antd/lib/list";
import { memo } from "react";
import { Event } from "../../../features/eventsSlice";
import { toLocalDate } from "../../../utils/date/dateHelper";

interface EventProps {
  item: Event;
}

const EventItem: React.FC<EventProps> = ({ item }) => {
  return (
    <List.Item>
      <List.Item.Meta title={item.name} description={item.description} />
      {`From ${toLocalDate(item.startDate)} to ${toLocalDate(item.endDate)}`}
    </List.Item>
  );
};

export default memo(EventItem);
