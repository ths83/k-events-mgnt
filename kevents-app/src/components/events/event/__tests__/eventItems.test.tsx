import React from "react";
import renderer from "react-test-renderer";
import { mockEvent } from "../../../../fixtures/EventsFixtures";
import EventItem from "../EventItem";

test("renders component", () => {
  const item = renderer.create(<EventItem item={mockEvent} />).toJSON();
  expect(item).toMatchSnapshot();
});
