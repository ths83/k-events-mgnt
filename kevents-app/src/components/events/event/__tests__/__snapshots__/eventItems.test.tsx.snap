// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<li
  className="ant-list-item ant-list-item-no-flex"
>
  <div
    className="ant-list-item-meta"
  >
    <div
      className="ant-list-item-meta-content"
    >
      <h4
        className="ant-list-item-meta-title"
      >
        event1
      </h4>
      <div
        className="ant-list-item-meta-description"
      >
        description1
      </div>
    </div>
  </div>
  From Saturday 2022-01-15 21:17 to Saturday 2022-01-15 21:17
</li>
`;
